bindings/python/test/test_arrow.py

```diff
@@ -325,6 +327,65 @@ def test_string_bool(self):
             ),
         )
 
+    def test_auto_schema(self):
```

Comment:
Let's also add a test with heterogeneous type for the same field, eg:
```python
[{"a": 1},  {"a": "2"},  {"a": True}, {"a": 4}]
```

The test should confirm that we infer from the first type and fill other types with null.